### ISSUE TITLE
[Question] Which is correct?

### ISSUE THREAD

**Issue by b4b4r07** [jhawthorn/fzy#10]
Thank you for this great product! This is awesome 👍 

BTY, I wonder these behaviors.

(A).
```console
$ cat file
ABC
abc
...
...
xyz
XYZ
$ cat file | fzy
>
ABC
abc
...
...
xyz
XYZ
```

(B). 
```console
$ cat file
ABC
abc
...
...
...
xyz
XYZ
$ cat file | fzy
>
...
abc
...
ABC
...
xyz
XYZ
```

(C).
```console
$ cat file
ABC
abc
...
...
...
...
xyz
XYZ
$ cat file | fzy
>
XYZ
abc
...
...
...
...
xyz
ABC
```

It is no use increasing dots'  lines. Which is correct? All? I hope ~~(C)~~ (A) behavior in any case. Thanks!

**Comment by jhawthorn** [comment]
If I understand your question correctly, (A) is the behaviour fzy has. If no search has been entered, fzy will show the results in the exact order they came in.

Thanks for the kind words.

**Comment by b4b4r07** [comment]
Thank you for quick reply. In fact, I noticed my mistake (NOT C, A is correct and I hope).

For now, (B) and (C) happens. Is it a bug?

**Comment by jhawthorn** [comment]
@b4b4r07 this is what I see. Do you see different?

![](http://i.hawth.ca/s/1J0dwk1g.png)

**Comment by b4b4r07** [comment]
@jhawthorn Yes, 

![fzy](https://cloud.githubusercontent.com/assets/4442708/15460493/8c1e8f6c-20ec-11e6-837c-31a4932e116d.gif)

and I tested on:

```console
$ uname -a
Darwin pc336.local 15.4.0 Darwin Kernel Version 15.4.0: Fri Feb 26 22:08:05 PST 2016; root:xnu-3248.40.184~3/RELEASE_X86_64 x86_64
```

```console
$ fzy --version
fzy 0.3 (c) 2014 John Hawthorn
```

**Comment by b4b4r07** [comment]
@jhawthorn Any ideas?

**Comment by natecavanaugh** [comment]
@jhawthorn I can also confirm that this happens for me as well on 0.4 with OSX (installed via Homebrew), though for me it's only C that happens.

**Comment by jhawthorn** [comment]
Thanks. I can't reproduce this, but I think I know why it is happening (sort is run on the empty query, and `qsort` isn't stable). I'll try to fix it soon.

**Comment by jhawthorn** [comment]
@natecavanaugh @b4b4r07 This should be fixed in the latest master by 3b23f5714e5d4625117c52988edd972d3587db23 and e6f83538bd42a39867753902d31f3b370f5aa1c6

**Comment by b4b4r07** [comment]
@jhawthorn Awesome! Thank you very much. It means a lot to me.
